src/Xrefcheck/Verify.hs

```diff
@@ -685,18 +703,30 @@ checkExternalResource Config{..} link
         Just u -> pure u
       let reqLink = case parsedUrl of
             Left (url, option) ->
-              runReq defaultHttpConfig $
-              req method url NoReqBody ignoreResponse option
+              runReq httpConfig $
+                req method url NoReqBody ignoreResponse option
             Right (url, option) ->
-              runReq defaultHttpConfig $
-              req method url NoReqBody ignoreResponse option
+              runReq httpConfig $
+                req method url NoReqBody ignoreResponse option
 
       let maxTime = Time @Second $ unTime ncExternalRefCheckTimeout * timeoutFrac
 
       mres <- liftIO (timeout maxTime $ void reqLink) `catch`
         (either throwError (\() -> return (Just ())) . interpretErrors)
       maybe (throwError $ ExternalResourceSomeError "Response timeout") pure mres
 
+    isTemporaryRedirectCode :: Int -> Bool
+    isTemporaryRedirectCode 302 = True
+    isTemporaryRedirectCode 303 = True
+    isTemporaryRedirectCode 307 = True
+    isTemporaryRedirectCode _   = False
+
+    isPermanentRedirectCode :: Int -> Bool
+    isPermanentRedirectCode 301 = True
+    isPermanentRedirectCode 308 = True
+    isPermanentRedirectCode _   = False
```

Comment:
Minor nitpick: it is less boilerplate-y to use `\case`, but feel free to leave the current version if you like this syntax. Or, maybe even shorter: use `elem`.